src/main/java/io/logz/jmx2graphite/Jmx2GraphiteConfiguration.java

```diff
@@ -121,6 +134,23 @@ private GraphiteProtocol getGraphiteProtocol(Config config) {
         return null;
     }
 
+    private void setFilterPatterns(Config config) {
+        try {
+            whiteListPattern = Pattern.compile(config.hasPath(Jmx2GraphiteJavaAgent.WHITE_LIST_REGEX) ?
```

Comment:
Instead of running through a regular expression for nothing in case the user hasn't set one, how about using Optional<Pattern> ? If empty, do nothing - saves CPU cycles on nothing, and more readable and clear. Same for blacklist